tests/cpp/test_multidevice_overlap.cpp

```diff
@@ -91,49 +101,54 @@ class OverlapTest : public MultiDeviceTest {
     // full unsharded input on each rank makes it possible to compute the
     // expected result locally, hence, this way of doing is convenient for
     // validating data correctness.
-    at::TensorOptions options =
-        at::TensorOptions().dtype(at::kFloat).device(communicator_->device());
-    auto ta_unsharded = at::randn(ta_unsharded_sizes, options);
-    auto tb_unsharded = at::randn(tb_unsharded_sizes, options);
-    ta_ = at::empty(ta_sizes, options);
-    ta_.copy_(ta_unsharded.select(1, my_device_index_));
-    tb_ = at::empty(tb_sizes, options);
-    tb_.copy_(tb_unsharded.select(0, my_device_index_));
+    auto cpu_options = at::TensorOptions().dtype(at::kFloat);
+    at::TensorOptions gpu_options = cpu_options.device(communicator_->device());
+
+    ta_unsharded_ = at::empty(ta_unsharded_sizes, cpu_options);
+    tb_unsharded_ = at::empty(tb_unsharded_sizes, cpu_options);
```

Comment:
What was wrong with using gpu_options? It's not immediately clear to me, so some clarification in code comment would help.

Reply:
The reason is what I wrote in the PR description, under the section "Improve memory footprint": those tensors are large and only used for validating data correctness, so we shouldn't use device memory for them. Let me add a comment in the code.